corehq/apps/domain/utils.py

```diff
@@ -111,3 +115,62 @@ def guess_domain_language(domain_name):
     domain = Domain.get_by_name(domain_name)
     counter = Counter([app.default_language for app in domain.applications() if not app.is_remote_app()])
     return counter.most_common(1)[0][0] if counter else 'en'
+
+
+def send_mock_repeater_payloads(repeater_id, payload_ids, email_id):
+    from corehq.motech.repeaters.models import Repeater, RepeatRecord
+    repeater = Repeater.get(repeater_id)
+    repeater_type = repeater.doc_type
+    payloads = dict()
+    headers = ['note']
+    result_file_name = "bulk-payloads-%s-%s-%s.csv" % (repeater.doc_type, repeater.get_id, datetime.datetime.utcnow())
+
+    def get_payload(payload_id):
+        dummy_repeat_record = RepeatRecord(
+            domain=repeater.domain,
+            next_check=datetime.datetime.utcnow(),
+            repeater_id=repeater.get_id,
+            repeater_type=repeater_type,
+            payload_id=payload_id,
+        )
+        payload_doc = repeater.payload_doc(dummy_repeat_record)
+        payload = Repeater(None).get_payload(dummy_repeat_record, payload_doc)
+        if isinstance(payload, dict):
+            return payload
+        else:
+            return json.loads(payload)
+
+    def populate_payloads():
+        for payload_id in payload_ids:
+            try:
+                payload = get_payload(payload_id)
+                payloads[payload_id] = payload
+                headers = list(set(headers + payload.keys()))
+            except Exception as e:
+                payloads[payload_id] = {'note': 'Could not generate payload, %s' % e.message}
+
+    def create_result_file():
+        with open(result_file_name, 'w') as csvfile:
```

Comment:
Is this typically how we handle tempfiles?  I thought they were stored somewhere in particular - this will save it in the root of the repo.

Reply:
It looks like `DownloadBase` uses a tempfile:
https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/soil/__init__.py#L67
```python
In [1]: from tempfile import mkstemp

In [2]: mkstemp(suffix='.txt')
Out[2]: (10, '/tmp/tmpEAk0C8.txt')
```
I'm not totally sure of the distinction, except that tempfiles are sometimes wiped, and that files in the repo root are wiped on deploy.  I'd go with whatever `DownloadBase` uses, since that's been vetted.  Could be better if we run into disk space issues, and it may be safer if this occurs during a deploy.